float64 cx1
float64 cx2
float64 area
bool appear
float64 w_h_ratio
bool left_excess
bool right_excess
